### ISSUE TITLE
Saved XGBoost model returning different predictions

### ISSUE THREAD

**Issue by replacementAI** [dmlc/xgboost#7506]
my current code:
`model = XGBRegressor(seed=X, alpha=X, colsample_bylevel=X, colsample_bynode=X, colsample_bytree=X, eta=X, gamma=X, reg_lambda=X, max_depth=X, min_child_weight=X, subsample=X, n_estimators=X)
`
`model.fit(X, y)`
`model.save_model('savedModel12-7')`
how I am loading the model with new information:
`model = XGBRegressor(seed=X, alpha=X, colsample_bylevel=X, colsample_bynode=X, colsample_bytree=X, eta=X, gamma=X, reg_lambda=X, max_depth=X, min_child_weight=X, subsample=X, n_estimators=X)
`
`model.load_model('savedModel12-7')`
`model.fit(X,y)`
`predicted=model.predict(X.tail(1))`
but that gives a different prediction if I comment out the model.fit(X,y) line, so I want to know what is the proper way of loading a saved model. Thanks!

**Comment by hcho3** [comment]
It might be a bug. Would you be able to share the set of hyperparameters and data (if possible) so that we can try to troubleshoot it?

**Comment by replacementAI** [comment]
> It might be a bug. Would you be able to share the set of hyperparameters and data (if possible) so that we can try to troubleshoot it?

sure but one question, am I supposed to refit the saved model when I load it, and then run the new X thru the predictor? Or am I supposed to just not fit the model, load the saved model, and run the new X thru the predictor?

**Comment by hcho3** [comment]
You won't need to re-fit when loading the saved model.

**Comment by replacementAI** [comment]
> You won't need to re-fit when loading the saved model.

Thanks, thats what was causing the difference in prediction outputs.
